plugins/content/caption/base/lib/captionManagers/srtCaptionsContentManager.php

```diff
@@ -139,4 +139,40 @@ public function merge($content, $toAppend)
 	{
 		return $content . $toAppend;
 	}
-}
\ No newline at end of file
+
+	public function createCaptionsFile($content, $clipStartTime, $clipEndTime, $timeCode, $globalOffset)
+	{
+		static $newLineIndex = 1;
+		$newFileContent = '';
+		$originalFileContentArray = kCaptionsContentManager::getFileContentAsArray($content);
+		while (($line = kCaptionsContentManager::getNextValueFromArray($originalFileContentArray)) !== false)
+		{
+			$currentBlock = '';
+			$shouldAddBlockToNewFile = true;
+			while (trim($line) !== '' and $line !== false)
+			{
+				$matches = array();
+				$timecode_match = preg_match($timeCode, $line, $matches);
+				if ($timecode_match)
+				{
+					$adjustedTimeLine = $this->createAdjustedTimeLine($matches, $clipStartTime, $clipEndTime, $globalOffset);
+					if($adjustedTimeLine)
+						$currentBlock .=$adjustedTimeLine;
+					else
+						$shouldAddBlockToNewFile = false;
+				}
+				else
+					$currentBlock .= $line . kCaptionsContentManager::UNIX_LINE_ENDING;
+				$line = kCaptionsContentManager::getNextValueFromArray($originalFileContentArray);
+			}
+			if($shouldAddBlockToNewFile)
```

Comment:
No need to copy the entire code.
Just refactor createCaptionsFile on class kCaptionsContentManager
And only this code-block
AddBlockToNewFile needs to be override in the class of srtCaptionsContentManager.php.